### DISCUSSION TITLE
[@types/three] `loadAsync` method missing for all `Loader` classes

### DISCUSSION THREAD

**Discussion by musjj** [DefinitelyTyped/DefinitelyTyped#65600] [Issues with a @types package] [answered]
The `Loader` class contains a `loadAsync` method which works exactly like `load`, but it returns a Promise.

This method is inherited by all `Loader` classes, e.g. `ImageLoader`, `TextureLoader`, `FontLoader`, etc.

Docs: https://threejs.org/docs/index.html#api/en/loaders/Loader.loadAsync
Implementation: https://github.com/mrdoob/three.js/blob/70cc4e192fe2ebd0bf8542a81c8c513d61984c58/src/loaders/Loader.js#L19

[a comment by an automated account is not shown]

**Comment by Methuselah96** [comment, marked as answer]
Thanks for the issue. It's most likely to get responded to if you file it at https://github.com/three-types/three-ts-types since that's where we maintain the types.

That being said, you should find that each loader has a `loadAsync()` method. The declaration types for the base `Loader` class do not define a `loadAsync()` method because the type for the return of `loadAsync()` depends on the loader. This could be solved by making `Loader` generic, but that work has not been done yet.

**Reply by musjj** [reply]
You're right, looks like the problem is with the base `Loader`. I was using the loaders from `three-stdlib` (e.g. [FontLoader](https://github.com/pmndrs/three-stdlib/blob/main/src/loaders/FontLoader.ts)), and they extend from the `Loader` base class. I guess that's why it was missing the `loadAsync` method.

**Comment by musjj** [comment]
I'll close this now, since it was a `three-stdlib` issue and it has been fixed: https://github.com/pmndrs/three-stdlib/pull/258
